src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unnecessaryparentheses/InputUnnecessaryParenthesesIfStatement.java

```diff
@@ -0,0 +1,71 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.unnecessaryparentheses;
+
+/* Config: default
+ */
+
+public class InputUnnecessaryParenthesesIfStatement {
+
+    void method(String sectionName) {
+        if ("Content".equals(sectionName) || "Overview".equals(sectionName)
+                || (!"AbbreviationAsWordInName".equals(sectionName) // violation, unnecessary
+                                                                    // parenthesis
+                && !"AbstractClassName".equals(sectionName) // ok
+        )) {
+            System.out.println("sd");
+        }
+    }
+
+    private void method() {
+        int x, y, z;
+        x = 0;
+        y = 0;
+
+        z = (x < y) ? x : y; // ok
+
+        if ((x < y) && (x > z)) { // ok
+            return;
+        }
+
+        if (((x < y) && (x > z))) { // violation, unnecessary parenthesis
+            return;
+        }
+
+        if (!(x <= y) || (x >= z)) { // ok
```

Comment:
this is following the pattern `!a || (b)` and does not have a violation.

I am confused by these and would like some clarification.

Reply:
fixed now a violation